### PR TITLE
fixes flaky visual tests

### LINKED ISSUE #22994: Fix flaky visual tests
Some of our visual tests has become flaky with new features added because the code do not wait for all requests to be resolved. We need to detect flaky visual tests and make them stable.

### PR DESCRIPTION
**Pull request by alxnddr**
Closes https://github.com/metabase/metabase/issues/22994

## Changes

Fixes most of flaky visual tests except ones where BE data changes cause visual diff

[Percy diff](https://percy.io/6a8dd5c9/metabase/builds/18950036?utm_campaign=6a8dd5c9&utm_content=metabase&utm_source=github_status_private)

### CODE REVIEW

File: frontend/test/metabase-visual/notebook/notebook.cy.spec.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ describe("visual tests > notebook > Run buttons", () => {
     cy.findByText("SQL query").click();
 
     // Check that we're on the blank question page
-    cy.findByText("Here's where your results will appear");
+    cy.findByText("Here's where your results will appear").click();
```
Comment (kulyk) [inline frontend/test/metabase-visual/notebook/notebook.cy.spec.js:95]: Is this copy clickable at all? 🤔

File: frontend/test/metabase-visual/models/editor.cy.spec.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ describe("visual tests > models > editor", () => {
         cy.visit(`/model/${MODEL_ID}/query`);
         cy.wait("@cardQuery");
         cy.findByText(/Doing science/).should("not.exist");
+        cy.wait(100); // waits for the colums widths calculation
```
Comment (kulyk) [inline frontend/test/metabase-visual/models/editor.cy.spec.js:23]: @nemanjaglumac do you happen to know a way we can avoid `wait` usages here?

File: frontend/test/metabase-visual/dashboard/parameters-widget.cy.spec.js [hunk 3]
```diff
@@ -183,14 +180,12 @@ describe(`visual tests > dashboard > parameters widget`, () => {
...
-        cy.findByText("test question");
-
-        cy.viewport(375, 667); // iPhone SE
+        cy.viewport(MOBILE_WIDTH, 667);
 
         cy.get("main")
           .scrollTo(0, 264)
           .then(() => {
-            cy.percySnapshot();
+            cy.percySnapshot(null, { widths: [MOBILE_WIDTH] });
```
Comment (kulyk) [inline frontend/test/metabase-visual/dashboard/parameters-widget.cy.spec.js:188]: Thanks for adding mobile snapshots too, nice!

**Review by kulyk — APPROVED** [review]

File: frontend/test/metabase-visual/notebook/notebook.cy.spec.js (hunk 1 above)
Comment (alxnddr) [reply frontend/test/metabase-visual/notebook/notebook.cy.spec.js:95]: > Is this copy clickable at all? 🤔

Nope, I added it to cause blur on the ace editor which did not work through getting the node and triggerting a blur event

File: frontend/test/metabase-visual/models/editor.cy.spec.js (hunk 2 above)
Comment (alxnddr) [reply frontend/test/metabase-visual/models/editor.cy.spec.js:23]: I did not find a convenient way to wait for table finishes resizing which can be async 😕

**Pull request merged**